ouroboros-consensus/docs/diffusion-pipelining-design/block-diffusion-pipelining.tex

```diff
@@ -0,0 +1,554 @@
+scaling linearly with the number of hops a block travels during its diffusion across the critical path.
+Let us now consider the following implementation changes to data diffusion in the Consensus layer to achieve
+a faster block propagation and adoption time with pipelining, the goal is to provide a \emph{mechanism} by which we 
+can expedite a block's propagation prior to its validation whilst not invalidating existing protocol primitives:
+
+\begin{itemize}
+  \item The aim is to expedite data propagation where appropriate, and to do so without destructive change.
+  \begin{enumerate}
+    \item Extend the current API to pre-notify subscribed threads when a new desirable block is seen, prior to block body validation.
+    \item Implement a new mini-protocol which can function as an "expedited" version of \lstinline{chainSync}.
+    \item The new protocol should provide two new mechanisms:
+    \begin{itemize}
+      \item The ability to send a fresh block header to a downstream peer as soon as a given node witnesses a new desirable block, prior to actually validating its body. This enables the downstream peer to pre-emptively fetch this new block.
+      \item The ability for a downstream peer to register interest in receiving another expedited block. 
+    \end{itemize}
+    \item Regular sanity checks still apply:
+    \begin{itemize}
+      \item Ignore headers which don't extend the currently selected chain.
+      \item Disconnect from a peer if a given header is invalid.
+      \item Do not expedite any data which is not definitively more desirable than all previous data expedited. 
+    \end{itemize} 
+  \end{enumerate}
+  \item Appropriately handle misbehaving peers, as per \cref{many-invalid-block-bodies}. Further discussion on this point is included in the section following, \cref{conclusion-and-considerations}.
+\end{itemize}
+  
+\pagebreak
+  
+\section{Conclusion \& considerations}
+\label{conclusion-and-considerations}
+It appears that diffusion pipelining is an exciting and potentially highly valuable 
+modification to the block propagation approach for Cardano core. There are some considerations
+though which one should bear in mind during implementation:
+
+\begin{itemize}
+  \item We must of course, ensure faster than real-time syncing of the chain. This is not an issue for the proposed implementation herein, however it's something we must be cognisant of in the future.
+  \item The changes proposed herein don’t affect the structural properties of the chain.
+  \item Intuitively one may be inclined to suggest a change to minting policy, given that with pipelining an SPO now \emph{knows} that a potentially better chain exists to extend. However, knowledge of this reality (\emph{which existed prior to pipelining}), should not alter minting behaviour. 
+  \item We should not expedite/pipeline a subsequent block until the previously pipelined block is validated, i.e. do not pipeline multiple blocks concurrently.
```

Comment:
I think it'd help to have a second sentence here explaining why we don't think this would be worth the extra complexity.

Reply:
agreed!